x/programs/wasmlanche/src/context.rs

```diff
@@ -209,23 +245,11 @@ impl Context {
     /// # Panics
     /// Panics if there was an issue deserializing the account
     #[must_use]
-    pub fn deploy(&self, program_id: ProgramId, account_creation_data: &[u8]) -> Address {
-        #[link(wasm_import_module = "program")]
-        extern "C" {
-            #[link_name = "deploy"]
-            fn deploy(ptr: *const u8, len: usize) -> HostPtr;
-        }
-
-        match self {
-            Context::Injected(_) => {}
-            #[cfg(feature = "bindings")]
-            Context::External(_) => panic!("not supported"),
-        }
-
+    pub fn deploy(&mut self, program_id: ProgramId, account_creation_data: &[u8]) -> Address {
         let ptr =
             borsh::to_vec(&(program_id, account_creation_data)).expect("failed to serialize args");
 
-        let bytes = unsafe { deploy(ptr.as_ptr(), ptr.len()) };
+        let bytes = self.host_accessor_mut().deploy(ptr.as_ptr(), ptr.len());
```

Comment:
🤔, the `host_accessor` should be immutable. We can use interior mutability in `cfg(feature = "test")`